### ISSUE TITLE
apprequest bug on "askfor", dialog not showing

### ISSUE THREAD

**Issue by tbase82** [jeduan/cordova-plugin-facebook4#521]
Hi,
if i use the following object for showDialog:
{
	method: "apprequests",
	message: "Message",
	title: 'Title',
	actionType: 'askfor',
	filters: 'app_non_users'
}

the dialog is not showing.

xcode output:
UserInfo={com.facebook.sdk:FBSDKErrorDeveloperMessageKey=The objectID is required when the actionType is either send or askfor., com.facebook.sdk:FBSDKErrorArgumentNameKey=objectID}
2017-06-17 23:45:20.142985+0200 OkeyAlemi[5596:1657936] game request dialog did fail
2017-06-17 23:45:20.143238+0200 OkeyAlemi[5596:1657936] error::Error Domain=com.facebook.sdk.share Code=2 "(null)" 

after changing the object id there is still an error.

my installed plugins:
$ cordova plugin list
cc.fovea.cordova.purchase 6.1.2 "Purchase"
cordova-plugin-compat 1.1.0 "Compat"
cordova-plugin-customurlscheme 4.3.0 "Custom URL scheme"
cordova-plugin-device 1.1.6 "Device"
cordova-plugin-facebook4 1.9.1 "Facebook Connect"
cordova-plugin-file 4.3.3 "File"
cordova-plugin-insomnia 4.3.0 "Insomnia (prevent screen sleep)"
cordova-plugin-screen-orientation 2.0.1 "Screen Orientation"
cordova-plugin-splashscreen 4.0.3 "Splashscreen"
cordova-plugin-statusbar 2.2.3 "StatusBar"
cordova-plugin-vibration 2.1.5 "Vibration"
cordova-plugin-whitelist 1.3.2 "Whitelist"
cordova-plugin-wkwebview-engine 1.1.3 "Cordova WKWebView Engine"
phonegap-plugin-push 1.10.4 "PushPlugin"
{}
$ cordova --version
7.0.1

if i change the actionType: to 'turn' its working for type turn. 
**If i put a empty string actionType: '' it works fine.**

Login with Facebook works fine.

compiled in debug mode and tested on a real device iphone7 with version 10.3.1

**Comment by fritio833** [comment]
Rolling back to facebook4 version 1.7.4 fixed the problem for me.

**Comment by bnfrnz** [comment]
+1
I have the same issue. In the [Facebook docs](https://developers.facebook.com/docs/games/services/gamerequests#dialogparameters) it says that action_type can only be used with object_id, and vice versa. 

I tried leaving both out when calling showDialog but then nothing happens. Only after reading above comment I got it to work (thanks @tbase82 !). 

But I assume this empty string solution should be considered a workaround and this could still use a proper fix.
